File: solutions/julia/saddle-points/2/saddle-points.jl
function saddlepoints(M)
    result = []
    no_rows = size(M,1)
    no_cols = size(M,2)
    for r in 1:no_rows
        m = maximum(M[r,1:no_cols])
        for c in 1:no_cols
            if M[r,c]==m && minimum(M[1:no_rows,c])==m
                push!(result,(r,c)) end            
        end
     end
    result
end

File: solutions/python/saddle-points/2/saddle_points.py
def saddle_points(matrix):
    result = []
    if matrix==[]:
        return result
    if any(len(row)!=len(matrix[0]) for row in matrix):
        raise ValueError("irregular matrix")
    no_rows = len(matrix)
    no_cols = len(matrix[0])
    for r in range(0, no_rows):
        m = max(matrix[r])        
        for c in range(0, no_cols):
            if matrix[r][c]==m and min([row[c] for row in matrix])==m:
              result.append({"row" : r+1, "column" : c+1})  
    return result